circle.yml

```diff
@@ -1,14 +1,44 @@
-dependencies:
-  cache_directories:
-    - deps
-    - _build
-  pre:
-    - wget http://packages.erlang-solutions.com/erlang-solutions_1.0_all.deb && sudo dpkg -i erlang-solutions_1.0_all.deb
-    - sudo apt-get update
-    - sudo apt-get install elixir
-    - yes | mix deps.get
-    - mix local.rebar
-test:
-  override:
-    - mix test
-    - MIX_ENV=prod mix compile
+# Elixir CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-elixir/ for more details
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/elixir:1.6.2
+        environment:  # environment variables for primary container
+          MIX_ENV: test
+    working_directory: ~/repo
+    steps:  # commands that comprise the `build` job
+      - checkout  # check out source code to working directory
+      - run: mix local.hex --force  # install Hex locally (without prompt)
+      - run: mix local.rebar --force  # fetch a copy of rebar (without prompt)
+      - restore_cache:  # restores saved mix cache
+          keys:  # list of cache keys, in decreasing specificity
+            - v1-mix-cache-{{ checksum "mix.lock" }}
+            - v1-mix-cache-{{ .Branch }}
+            - v1-mix-cache
+      - restore_cache:  # restores saved build cache
+          keys:
+            - v1-build-cache-{{ .Branch }}
+            - v1-build-cache
+      - run: mix do deps.get, compile  # get updated dependencies & compile them
+      - save_cache:  # generate and store cache so `restore_cache` works
+          key: v1-mix-cache-{{ checksum "mix.lock" }}
+          paths: "deps"
+      - save_cache:  # make another less specific cache
+          key: v1-mix-cache-{{ .Branch }}
+          paths: "deps"
+      - save_cache:  # you should really save one more cache just in case
+          key: v1-mix-cache
+          paths: "deps"
+      - save_cache: # don't forget to save a *build* cache, too
+          key: v1-mix-cache-{{ checksum "mix.lock" }}
+          paths: "_build"
+      - save_cache: # don't forget to save a *build* cache, too
+          key: v1-build-cache-{{ .Branch }}
+          paths: "_build"
+      - save_cache: # and one more build cache for good measure
+          key: v1-build-cache
+          paths: "_build"
```

Comment:
Could you explain all these caches though before I merge in? I'm not sure why another cache is needed for _build and deps

Reply:
To be honest I'm not 100% sure you even need that. I was just going off their site at 
https://circleci.com/docs/2.0/language-elixir/

That being said I see you merged the other PR for circle ci I can just drop that from this.

Reply:
oh wait, sorry I thought you had. 

Well, from what I can read. It looks like its a base cache for build vs the other { .Branch } one. 

I have used this config on other projects and it does not look to be an issue running in terms of resources. Though I really can't say if you will see less or more performance dropping the caches.